### PR TITLE
issue #297 Document versioning strategy

### PR DESCRIPTION
**Pull request by wajda**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,8 +41,26 @@ The agent can be used with or without a Spline server, depending on your use cas
...
+In other words, the _Spline Spark Agent Public API_ in terms of _SemVer_ covers all entities and abstractions that are designed 
+to be used or extended by client applications.
+
+The version number **does not** directly reflect the relation of the Agent to the Spline Producer API (the Spline server). Both the Spline Server and
+the Agent are designed to be as much mutually compatible as possible, assuming long-term operation and a possibly significant gap in the server and
+the agent release dates. Such requirement is dictated by the nature of the Agent that could be embedded into some Spark jobs and only rarely if ever
+updated without posing a risk to stop working because of eventual Spline server update. Likewise, it should be possible to update the Agent anytime
+(e.g. to fix a bug or support a newer Spark version or a feature that earlier agent version didn't support) without requiring a Spline server upgrade.
+
+Although isn't required (see above), but for minimizing user astonishment when and if a compatibility between too distant versions of the Agent and
```
Comment (cerveada) [inline README.md]: `Although isn't required (see above), ` you mean required by SemVer? Also `it` is missing.
`when and if` - why not just `when`? is there a difference?

Comment (wajda) [reply README.md]: rephrased

File: README.md [hunk 2]
```diff
@@ -40,8 +41,26 @@ The agent can be used with or without a Spline server, depending on your use cas
...
+to be used or extended by client applications.
+
+The version number **does not** directly reflect the relation of the Agent to the Spline Producer API (the Spline server). Both the Spline Server and
+the Agent are designed to be as much mutually compatible as possible, assuming long-term operation and a possibly significant gap in the server and
+the agent release dates. Such requirement is dictated by the nature of the Agent that could be embedded into some Spark jobs and only rarely if ever
+updated without posing a risk to stop working because of eventual Spline server update. Likewise, it should be possible to update the Agent anytime
+(e.g. to fix a bug or support a newer Spark version or a feature that earlier agent version didn't support) without requiring a Spline server upgrade.
+
+Although not required by the above statement, for minimizing user astonishment when the compatibility between too distant the _Agent_ and the _Server_
+versions is dropped, we'll increment the _Major_ version component.
```
Comment (cerveada) [inline README.md]:
```suggestion
Although not required by the above statement, for minimizing user astonishment when the compatibility between too distant _Agent_ and _Server_
versions is dropped, we'll increment the _Major_ version component.
```

**Review by cerveada — DISMISSED** [review]
See the last comment, the rest LGTM

**Pull request merged**